R/fullVALIDATIONREMIND.R

```diff
@@ -17,42 +17,58 @@ fullVALIDATIONREMIND <- function(rev = 0) {
   # Determines all regions data should be aggregated to by examining the columns
   # of the `regionmapping` and `extramappings` currently configured.
 
-  rel <- 'global'   # always compute global aggregate
+  rel <- "global"   # always compute global aggregate
   for (mapping in c(getConfig("regionmapping"), getConfig("extramappings"))) {
-    columns <- setdiff(colnames(toolGetMapping(mapping, 'regional')),
-                       c('X', 'CountryCode'))
+    columns <- setdiff(colnames(toolGetMapping(mapping, "regional")),
+                       c("X", "CountryCode"))
 
-    if (any(columns %in% rel))
-      warning("The following column(s) from ", mapping,
-              " exist in another mapping an will be ignored: ",
-              paste(columns[columns %in% rel], collapse = ", "))
+    if (any(columns %in% rel)) {
+      warning(
+        "The following column(s) from ", mapping,
+        " exist in another mapping an will be ignored: ",
+        paste(columns[columns %in% rel], collapse = ", ")
+      )
+    }
 
     rel <- unique(c(rel, columns))
   }
 
-  columnsForAggregation <- gsub("RegionCode", "region",
-                                paste(rel, collapse = "+"))
+  columnsForAggregation <- gsub(
+    "RegionCode", "region",
+    paste(rel, collapse = "+")
+  )
 
   # historical data ----
   valfile <- "historical.mif"
 
-  calcOutput("Historical", round = 5,  file = valfile, aggregate = columnsForAggregation,
-             append = FALSE, warnNA = FALSE, try = TRUE)
+  calcOutput("Historical",
+    round = 5, file = valfile, aggregate = columnsForAggregation,
+    append = FALSE, warnNA = FALSE, try = TRUE
+  )
 
-  calcOutput(type = "IEA_ETP", aggregate = "global", file = valfile,
-             append = TRUE, warnNA = FALSE, try = TRUE, isValidation = TRUE)
+  calcOutput(
+    type = "IEA_ETP", aggregate = columnsForAggregation, file = valfile,
+    append = TRUE, warnNA = FALSE, try = TRUE, isValidation = TRUE
+  )
 
-  calcOutput(type = "IEA_ETP", aggregate = "region", file = valfile,
-             append = TRUE, warnNA = FALSE, try = TRUE, isValidation = TRUE)
+  calcOutput(
+    type = "IEA_WEO_2021", subtype = "global", aggregate = columnsForAggregation,
+    file = valfile,append = TRUE, warnNA = FALSE, try = TRUE, isValidation = TRUE
+  )
 
-  calcOutput(type = "IEA_WEO_2021", aggregate = "global", file = valfile,
-             append = TRUE, warnNA = FALSE, try = TRUE, isValidation = TRUE)
-
-  calcOutput(type = "IEA_WEO_2021", aggregate = "region", file = valfile,
-             append = TRUE, warnNA = FALSE, try = TRUE, isValidation = TRUE)
+  calcOutput(
```

Comment:
An unresolved problem. calcIEA_WEO_2021 is called twice with 2 different subtypes. What we want is to remove all variables for any regions other than GLO from the calculation with subtype = "global".  And we want to remove all variables for GLO from the calculation with subtype = "region".  Cannot be resolved with the current approach, which indeed creates duplicates in the historical.mif as is.

Reply:
One option is to do it here before writing to historical.mif right away.

Reply:
Actually had to implement another solution, which in the case of IEA WEO 2021 filters before appending to historical.mif.